File: server/src/db/schemas/FriendRequest.ts
import { Schema } from 'mongoose';

export const friendRequestSchema = new Schema({
  fromId: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  toId: {
    type: Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
});

friendRequestSchema.index({ fromId: 1, toId: 1 }, { unique: true });

File: mobile/latticechat/lib/logic/models/friend_request.dart
class FriendRequestModel {
  final String id;
  final String fromId;
  final String toId;

  FriendRequestModel._({
    required this.id,
    required this.fromId,
    required this.toId
  });

  static FriendRequestModel fromJson(Map<String, dynamic> json) {
    return FriendRequestModel._(
        id: json['_id'],
        fromId: json['fromId'],
        toId: json['toId']
    );
  }
}